gulpfile.js

```diff
@@ -224,23 +224,216 @@ gulp.task("serve:theme-fork", function() {
+      {
+        test:    /.js?$/,
+        loader:  "babel",
+        exclude: /node_modules/
+      }
+    ]
+  }
+};
+
+gulp.task("build:theme:webpack:generate-development-js", function() {
+  return gulp.plumbedSrc(`${paths.src}/Js/Index.js`)
+    .pipe(webpackStream(Object.assign({}, commonWebpackConfig, {
+      watch: true
+    })))
+    .pipe(gulp.dest(`${paths.core}/Js`))
+    .pipe(livereload());
+});
+
+gulp.task("build:theme:webpack:generate-production-js", function() {
+  return gulp.src(`${paths.src}/Js/Index.js`)
+    .pipe(webpackStream(Object.assign({}, commonWebpackConfig, {
+      plugins: [
+        new webpack.optimize.UglifyJsPlugin({
+          compress: {
+            warnings: false
+          }
+        }),
+        new webpack.DefinePlugin({
+          'process.env.NODE_ENV': '"production"'
+        })
+      ]
+    }, webpack)))
+    .pipe(gulp.dest(`${paths.core}/Js`));
+});
+
+gulp.task("build:theme:assets:copy-templates", function() {
+  return gulp.plumbedSrc(`${paths.src}/Layout/Templates/**/*`)
+    .pipe(gulp.dest(`${paths.core}/Layout/Templates`))
+    .pipe(livereload());
+});
+
+gulp.task("build:theme:images:minify-images", function() {
+  return gulp.plumbedSrc(`${paths.src}/Layout/Images/**/*`)
+    .pipe(imagemin())
+    .pipe(gulp.dest(`${paths.core}/Layout/Images`))
+    .pipe(livereload());
+});
+
+gulp.task("build:theme", ["build:theme:empty-destination-folders"], function() {
+  gulp.start(
+    "build:assets:copy-images-vendors",
+    "build:theme:fonts:generate-iconfont",
+    "build:theme:fonts:generate-webfonts",
+    "build:theme:sass:generate-production-css",
+    "build:theme:webpack:generate-production-js",
+    "build:theme:assets:copy-templates",
+    "build:theme:images:minify-images"
+  );
+});
+
+gulp.task("serve:theme", function() {
+  livereload.listen();
+  gulp.watch(`${paths.src}/Js/**/*.js`, ["build:theme:webpack:generate-development-js"]);
+  gulp.watch(`${paths.src}/Layout/Sass/**/*.scss`, ["build:theme:sass:generate-development-css"]);
+  gulp.watch(`${paths.src}/Layout/Templates/**/*`, ["build:theme:assets:copy-templates"]);
+  gulp.watch(`${paths.src}/Layout/Images/**/*`, ["build:theme:images:minify-images"]);
+  gulp.watch(`${paths.src}/Layout/icon-sources/*`, ["build:theme:fonts:generate-iconfont"]);
+  gulp.watch(`${paths.src}/Layout/Fonts/**/*`, ["build:theme:fonts:generate-webfonts"]);
+});
+
 // public tasks
 gulp.task("default", function() {
   gulp.start("build");
 });
 
-gulp.task("serve", function() {
+gulp.task("serve:all", function() {
   gulp.start(
       "serve:backend",
       "serve:frontend",
-      "serve:theme-fork"
+      "serve:theme-fork",
+      "serve:theme"
   );
 });
 
-gulp.task("build", function() {
+gulp.task("build:all", function() {
   gulp.start(
       "build:backend",
       "build:frontend",
-      "build:theme-fork"
+      "build:theme-fork",
+      "build:theme" // @remark custom for SumoCoders
+  );
+});
+
+gulp.task("serve", function() {
```

Comment:
And why altering the core-stuff? Why not leave it `serve`, and let it watch "everything".
I'm just not a fan of changing the core-behaviour